server/controller/routes/user/registeration.js

```diff
@@ -0,0 +1,163 @@
+const yup = require('yup');
+const nodemailer = require('nodemailer');
+const ical = require('ical-generator');
+const moment = require('moment');
+
+const { checkUserExist } = require('../../../database/queries/users');
+const { getEventDetails } = require('../../../database/queries/events');
+const { getUsersCode } = require('../../../database/queries/users');
+const { signUserAttend } = require('../../../database/queries/users');
+const { alreadyBooked } = require('../../../database/queries/events');
+
+const checkUser = (req, res, next) => {
+  const mobileRegExp = /^(\+?\d{0,4})?\s?-?\s?(\(?\d{3}\)?)\s?-?\s?(\(?\d{3}\)?)\s?-?\s?(\(?\d{4}\)?)?$/;
+  const schema = yup.object().shape({
+    mobile: yup.string().matches(mobileRegExp),
+    eventCode: yup
+      .number()
+      .required()
+      .positive()
+      .integer()
+      .min(100)
+      .max(999),
+  });
+  const result = schema.isValidSync(req.body);
```

Comment:
Not a big issue but try to avoid using Sync apis in node code. It's not necessary to change now but this is a place where you could use async / await with the promise that `yup` returns